### PR TITLE
fix: props spreading to Menu

### PR DESCRIPTION
**Pull request by chrismanciero**
Add props spreading to the Menu component

### CODE REVIEW

File: src/Menu/Menu.js [hunk 1]
```diff
@@ -48,23 +51,26 @@ export const MenuItem = ({ url, link, isLink, separator, addon, children, onclic
...
+    linkProps: PropTypes.object,
     separator: PropTypes.bool,
     url: PropTypes.string
 };
 
 // ---------------------------------------- Menu Group ----------------------------------------
-export const MenuGroup = ({ title, children, className, ...props }) => {
+export const MenuGroup = ({ title, children, className, titleProps, ...props }) => {
     return (
         <div className={`fd-menu__group${className ? ' ' + className : ''}`} {...props}>
```
Comment (greg-a-smith) [inline src/Menu/Menu.js]: Although not changed, this should probably move the spread to be first.

File: src/Menu/Menu.test.js [hunk 2]
```diff
@@ -131,16 +131,56 @@ describe('<Menu />', () => {
...
         });
 
-        xtest('should allow props to be spread to the MenuItem component\'s a element', () => {
-            // TODO: placeholder for this test description once that functionality is built
+        test('should allow props to be spread to the MenuItem component\'s a element', () => {
+            const element = mount(
+                <MemoryRouter>
+                    <Menu>
+                        <MenuList>
+                            <MenuItem linkProps={{ 'data-sample': 'Sample' }} url='/' />
```
Comment (greg-a-smith) [inline src/Menu/Menu.test.js]: Per comment above, should this be a new prop (`urlProps`)?

File: src/Menu/Menu.js (hunk 1 above)
Comment (chrismanciero) [reply src/Menu/Menu.js]: Done

File: src/Menu/Menu.js [hunk 3]
```diff
@@ -22,24 +22,27 @@ export const MenuList = ({ children, className, ...props }) => {
 };
 
 // ---------------------------------------- Menu Item ----------------------------------------
-export const MenuItem = ({ url, link, isLink, separator, addon, children, onclick, className, ...props }) => {
+export const MenuItem = ({ url, link, isLink, separator, addon, children, onclick, className, addonProps, linkProps, ...props }) => {
     return (
         <React.Fragment>
             <li className={className} {...props}>
```
[the remark this answers is not in the document]
Comment (chrismanciero) [reply src/Menu/Menu.js]: Done

File: src/Menu/Menu.test.js (hunk 2 above)
Comment (chrismanciero) [reply src/Menu/Menu.test.js]: New prop - urlProps added

File: src/Menu/Menu.js [hunk 4]
```diff
@@ -22,24 +22,27 @@ export const MenuList = ({ children, className, ...props }) => {
...
                 {link &&
-                    <Link className={`fd-menu__item${isLink ? ' fd-menu__link' : ''}`} to={link}>
+                    <Link {...linkProps} className={`fd-menu__item${isLink ? ' fd-menu__link' : ''}`}
+                        to={link}>
                         {children}
                     </Link>
                 }
                 {url &&
-                    <a className={`fd-menu__item${isLink ? ' fd-menu__link' : ''}`} href={url}>
+                    <a {...linkProps} className={`fd-menu__item${isLink ? ' fd-menu__link' : ''}`}
```
[the remark this answers is not in the document]
Comment (chrismanciero) [reply src/Menu/Menu.js]: added new prop - urlProps for <a> elements

File: src/Menu/Menu.Component.js [hunk 5]
```diff
@@ -72,6 +72,18 @@ export const MenuComponent = () => {
...
+                        description: 'object - additional props to be spread to the addon section'
+                    },
+                    {
+                        name: 'linkProps',
+                        description: 'object - additional props to be spread to the Menu links'
+                    },
+                    {
+                        name: 'titleProps',
+                        description: 'object - additional props to be spread to the Menu Group title'
+                    },
```
[the remark this answers is not in the document]
Comment (chrismanciero) [reply src/Menu/Menu.Component.js]: How about something like this?

screen shot 2019-01-17 at 11 55 37 am

[the remark this answers is not in the document]
Comment (greg-a-smith) [reply src/Menu/Menu.Component.js]: I think that's a good start -- certainly good enough for this PR. 👍 I created #238 for a more widespread change across all component pages for a later date.

**Pull request merged**